### DISCUSSION TITLE
aio helm chart deployment hostPort conflicts with LoadBalencer definition

### DISCUSSION THREAD

**Discussion by drokath** [nextcloud/all-in-one#3433] [Questions] [answered]
### Steps to reproduce
1. Install the chart

### Expected behavior 
nextcloud-aio-apache pod should be running

### Actual behavior 
nextcloud-aio-apache pod is pending with the following message:
```
node(s) didn't have free ports for the requested pod ports. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
```

### Host OS 
NA

#### Nextcloud AIO version 
Helmchart 7.2.1

#### Current channel 
NA

**Comment by szaimen** [comment]
Hi, you can adjust the port via https://github.com/nextcloud/all-in-one/blob/f8a855f491a5ebd70e59e0e1b0b827298a274c03/nextcloud-aio-helm-chart/values.yaml#L22

**Reply by drokath** [reply]
Hi Simon,
Unfortunately the APACHE_PORT, is used both as a nodePort in deployment, and in the LoadBalancer which can not work on a single node. Hence the error.

On top of that, apache is exposed twice.

Just removing the nodePort definition would allow to run on a single node while exposing nextcloud-aio-apache only once.

**Reply by szaimen** [reply]
> On top of that, apache is exposed twice.

No, one is tcp and one udp

**Reply by szaimen** [reply]
> Unfortunately the APACHE_PORT, is used both as a nodePort in deployment, and in the LoadBalancer which can not work on a single node. Hence the error.

Are you sure about that? We use kompose to create the chart and it configures this automatically like this. See https://github.com/nextcloud/all-in-one/blob/f8a855f491a5ebd70e59e0e1b0b827298a274c03/nextcloud-aio-helm-chart/update-helm.sh#L46

**Comment by drokath** [comment]
Twice meaning as a nodePort for udp/top and LB for tcp/udp both use the
exact same port.

**Reply by szaimen** [reply]
> Twice meaning as a nodePort for udp/top and LB for tcp/udp both use the exact same port.

So can we get rid of the Loadbalancer then or how should we fix this in your opinion? The APACHE_PORT must be reachable from outside the cluster, this is why I think we need the loadbalancer?

**Comment by drokath** [comment]
Yes I am sure:

https://github.com/nextcloud/all-in-one/blob/main/nextcloud-aio-helm-chart/templates/nextcloud-aio-apache-deployment.yaml#L79

https://github.com/nextcloud/all-in-one/blob/main/nextcloud-aio-helm-chart/templates/nextcloud-aio-apache-service.yaml#L15-L16


Keeping the LB and removing the container hostPort would be better. As it
would ne easier to and an ingress through a reverse proxy from there.

**Comment by szaimen** [comment, marked as answer]
Will be fixed with https://github.com/nextcloud/all-in-one/pull/3445

**Comment by drokath** [comment]
Awesome thank you.

**Comment by szaimen** [comment]
Hi, this is now fixed with v7.4.1. Testing and feedback is welcome!
